### PR TITLE
Migration from unittests to pytest, remove /dev/shm

### PR DESCRIPTION
**Pull request by lobziik**
This PR introduces 2 big things and a few minor improvements and refactorings.

* Porting and fixing existing tests to the unittests infrastructure. This enables junit and coverage reports, as well as fixes return codes in case of tests failure. Temporary repo creation was also fixed, the current version does not work in OCP CI.

* Introducing `GithubAppWrapper` which now contains logic around `user_auth` parameter which previously was spreaded across the `rebasebot.bot` module code.
This class is intended to be instantiated right after the argument parsing step and passed into `bot.run` function, instead of a set of tokens. Now authenticated github_app instance might be obtained through this wrapper.

* Various small "improvements", like a bit of type hinting (where I felt strict necessity but still incomplete), tests runner script, and so on.

### CODE REVIEW

File: .pylintrc [hunk 1]
```diff
@@ -80,7 +80,11 @@ disable=broad-except,
         too-many-arguments,
         too-many-return-statements,
         too-many-branches,
-        too-many-statements
+        too-many-statements,
+
+        missing-function-docstring,
+        missing-module-docstring,
+        missing-class-docstring
```
Comment (JoelSpeed) [inline .pylintrc:87]: Disabling linters? How come?

Comment (lobziik) [reply .pylintrc:87]: Temporary. I would like to have other checks till I didn't write docstrings.

Comment (lobziik) [reply .pylintrc]: Enabled this back.

**Review by elmiko — COMMENTED** [review]
i did not test this locally, but in general i think it looks great. very nice improvements, and the type hinting feels very fancy :smiley_cat:

File: rebasebot/github.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/python
+
+# All Rights Reserved.
+#
```
[the remark this answers is not in the document]
Comment (lobziik) [reply rebasebot/github.py:4]: Hm... I just copied it from another file in this module to keep it consistent. Looking into the repo root, there is Apache license there already, think we can just remove this block.

File: rebasebot/github.py [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+from typing import Optional
+from collections import namedtuple
+
+import github3
+
+logger = logging.getLogger()
+
+
+class GitHubBranch(namedtuple("GitHubBranch", ["url", "ns", "name", "branch"])):
+    """
```
[the remark this answers is not in the document]
Comment (lobziik) [reply rebasebot/github.py:30]: No particular reason, just copy-pasting artifacts.
Fixed.

File: rebasebot/github.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (elmiko) [reply rebasebot/github.py]: ack, we should probably add copyright notices in another pr

**Review by Fedosin — APPROVED** [review]
/lgtm

**Pull request merged**